### PR TITLE
ROC-3022: Regenerate pin on support request

### PR DESCRIPTION
**Pull request by dharmendrak**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ROC-3022

### Change description ###
 Regenerate pin on support request

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/cmc/claimstore/controllers/SupportControllerTest.java [hunk 1]
```diff
@@ -118,17 +119,48 @@ public void shouldResendRPANotifications() {
...
 
+    @Test
+    public void shouldResendStaffNotifications() {
+        // given
+        sampleClaim = SampleClaim.builder()
+            .withClaimData(SampleClaimData.submittedByClaimant())
+            .withDefendantId(null)
+            .build();
+
+        List<String> sendList = new ArrayList<>();
```
Comment (andrewwa-kainos) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/controllers/SupportControllerTest.java]: This is not used

Comment (dharmendrak) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/controllers/SupportControllerTest.java]: removed.

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataService.java [hunk 2]
```diff
@@ -853,4 +854,50 @@ public void savePaidInFull(Long caseId, PaidInFull paidInFull, String authorisat
...
+                anonymousCaseWorker.getAuthorisation(),
+                eventRequestData,
+                caseId,
+                userDetails.isSolicitor() || userDetails.isCaseworker()
+            );
+
+            CCDCase ccdCase = extractCase(startEventResponse.getCaseDetails());
+            Claim claim = caseMapper.from(ccdCase);
+
+            Claim updatedClaim = claim.toBuilder()
```
Comment (karthikkannankainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataService.java]: why we are using claim.toBuilder(). The event should have the letterholderId ?

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataService.java [hunk 3]
```diff
@@ -853,4 +854,50 @@ public void savePaidInFull(Long caseId, PaidInFull paidInFull, String authorisat
...
+            EventRequestData eventRequestData = eventRequest(LINK_LETTER_HOLDER, userDetails.getId());
+
+            StartEventResponse startEventResponse = startUpdate(
+                anonymousCaseWorker.getAuthorisation(),
+                eventRequestData,
+                caseId,
+                userDetails.isSolicitor() || userDetails.isCaseworker()
+            );
+
+            CCDCase ccdCase = extractCase(startEventResponse.getCaseDetails());
```
Comment (karthikkannankainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataService.java]: The letterholderId should be passed as part of event.

File: src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/BaseIntegrationTest.java [hunk 4]
```diff
@@ -39,6 +39,7 @@
     protected static final String JURISDICTION_ID = "CMC";
     protected static final String CASE_TYPE_ID = "MoneyClaimCase";
     protected static final String CREATE_NEW_CASE = "IssueClaim";
+    protected static final String LINK_LETTER_HOLDER = "LinkLetterHolder";
```
Comment (karthikkannankainos) [inline src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/BaseIntegrationTest.java]: Dont recreate the event name. Please use the CaseEvent enum.

File: src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CCDCaseRepository.java [hunk 5]
```diff
@@ -174,6 +174,11 @@ public Claim saveClaimDocuments(
         return coreCaseDataService.saveClaimDocuments(authorisation, claimId, claimDocumentCollection);
     }
 
+    @Override
+    public Claim linkLetterHolder(Long claimId, String letterHolderId, String authorisation) {
```
Comment (karthikkannankainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CCDCaseRepository.java]: authorisation is not used at all, why are we passing?

File: src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/DBCaseRepository.java [hunk 6]
```diff
@@ -227,12 +227,23 @@ public void saveCaseEvent(String authorisation, Claim claim, CaseEvent caseEvent
...
+        String authorisation,
+        Long claimId,
+        ClaimDocumentCollection claimDocumentCollection
+    ) {
         claimRepository.saveClaimDocuments(claimId, jsonMapper.toJson(claimDocumentCollection));
+        return getClaimById(claimId);
+    }
+
+    @Override
+    public Claim linkLetterHolder(Long claimId, String letterHolderId, String authorisation) {
```
Comment (karthikkannankainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/DBCaseRepository.java]: Authorisation is not used. unnecessary.

File: src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CaseRepository.java [hunk 7]
```diff
@@ -64,5 +64,7 @@ void saveCountyCourtJudgment(
     void saveCaseEvent(String authorisation, Claim claim, CaseEvent caseEvent);
 
     Claim saveClaimDocuments(String authorisation, Long claimId, ClaimDocumentCollection claimDocumentCollection);
+
+    Claim linkLetterHolder(Long claimId, String letterHolderId, String authorisation);
```
Comment (karthikkannankainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CaseRepository.java]: Authorisation is not required. Remove it please

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataService.java (hunk 3 above)
Comment (dharmendrak) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataService.java]: it's our existing process of conversion from case to claim

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataService.java (hunk 2 above)
Comment (dharmendrak) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataService.java]: it's our existing process of conversion from case to claim

File: src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CCDCaseRepository.java (hunk 5 above)
Comment (dharmendrak) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CCDCaseRepository.java]: good pick

File: src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CaseRepository.java (hunk 7 above)
Comment (dharmendrak) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CaseRepository.java]: good pick

File: src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/DBCaseRepository.java (hunk 6 above)
Comment (dharmendrak) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/DBCaseRepository.java]: good pick

File: src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/BaseIntegrationTest.java (hunk 4 above)
Comment (dharmendrak) [reply src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/BaseIntegrationTest.java]: good pick, done

**Pull request merged**